darkdetect/_windows_detect.py

```diff
@@ -70,53 +73,50 @@ def theme():
-        ctypes.wintypes.DWORD(),
-        ctypes.wintypes.DWORD(0x00020019), # KEY_READ
-        ctypes.byref(hKey),
-    )
+class WindowsListener(BaseListener):
 
-    dwSize = ctypes.wintypes.DWORD(ctypes.sizeof(ctypes.wintypes.DWORD))
-    queryValueLast = ctypes.wintypes.DWORD()
-    queryValue = ctypes.wintypes.DWORD()
-    advapi32.RegQueryValueExA(
-        hKey,
-        ctypes.wintypes.LPCSTR(b'AppsUseLightTheme'),
-        ctypes.wintypes.LPDWORD(),
-        ctypes.wintypes.LPDWORD(),
-        ctypes.cast(ctypes.byref(queryValueLast), ctypes.wintypes.LPBYTE),
-        ctypes.byref(dwSize),
-    )
-
-    while True:
-        advapi32.RegNotifyChangeKeyValue(
-            hKey,
-            ctypes.wintypes.BOOL(True),
-            ctypes.wintypes.DWORD(0x00000004), # REG_NOTIFY_CHANGE_LAST_SET
-            ctypes.wintypes.HANDLE(None),
-            ctypes.wintypes.BOOL(False),
+    def _listen(self):
+        hKey = ctypes.wintypes.HKEY()
+        advapi32.RegOpenKeyExA(
+            ctypes.wintypes.HKEY(0x80000001), # HKEY_CURRENT_USER
+            ctypes.wintypes.LPCSTR(b'SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize'),
+            ctypes.wintypes.DWORD(),
+            ctypes.wintypes.DWORD(0x00020019), # KEY_READ
+            ctypes.byref(hKey),
         )
+
+        dwSize = ctypes.wintypes.DWORD(ctypes.sizeof(ctypes.wintypes.DWORD))
+        queryValueLast = ctypes.wintypes.DWORD()
+        queryValue = ctypes.wintypes.DWORD()
         advapi32.RegQueryValueExA(
             hKey,
             ctypes.wintypes.LPCSTR(b'AppsUseLightTheme'),
             ctypes.wintypes.LPDWORD(),
             ctypes.wintypes.LPDWORD(),
-            ctypes.cast(ctypes.byref(queryValue), ctypes.wintypes.LPBYTE),
+            ctypes.cast(ctypes.byref(queryValueLast), ctypes.wintypes.LPBYTE),
             ctypes.byref(dwSize),
         )
-        if queryValueLast.value != queryValue.value:
-            queryValueLast.value = queryValue.value
-            callback('Light' if queryValue.value else 'Dark')
+
+        while True:
+            advapi32.RegNotifyChangeKeyValue(
+                hKey,
+                ctypes.wintypes.BOOL(True),
+                ctypes.wintypes.DWORD(0x00000004), # REG_NOTIFY_CHANGE_LAST_SET
+                ctypes.wintypes.HANDLE(None),
+                ctypes.wintypes.BOOL(False),
+            )
+            advapi32.RegQueryValueExA(
+                hKey,
+                ctypes.wintypes.LPCSTR(b'AppsUseLightTheme'),
+                ctypes.wintypes.LPDWORD(),
+                ctypes.wintypes.LPDWORD(),
+                ctypes.cast(ctypes.byref(queryValue), ctypes.wintypes.LPBYTE),
+                ctypes.byref(dwSize),
+            )
+            if queryValueLast.value != queryValue.value:
+                queryValueLast.value = queryValue.value
+                self.callback('Light' if queryValue.value else 'Dark')
+
```

Comment:
I do not think we can avoid defining `_stop` and `_wait` for this class, as the user will (legitimately) expect those. I am fine defining `_wait` as just `_stop`. For `_stop`, we can either implement it in a way that just disables the callback, or dig in the Win32 API for the opposite of `RegNotifyChangeKeyValue`.

Reply:
I don't have access to a Windows outside of a VM nor am I familiar enough with the OS so this would be appreciated. Also, it suffices to define `_wait` as `pass`, as `wait` automatically calls `_stop` as needed in the base class.

Reply:
It's ok, for now I think it would suffice to implement `_stop` in a way that just stops the infinite loop and prevents any further call to the callback function is ok. I will dig in the Win32 API in the future, if needed (I think this is what we want: https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regclosekey).